tracer/src/Datadog.Trace/LibDatadog/TraceExporter.cs

```diff
@@ -59,6 +64,21 @@ public Task<bool> SendTracesAsync(ArraySegment<byte> traces, int numberOfTraces,
 #pragma warning restore DDLOG004
                         throw ex;
                     }
+                    else
+                    {
+                        if (responsePtr != IntPtr.Zero)
+                        {
+                            string? response = Marshal.PtrToStringAnsi(NativeInterop.ExporterResponse.GetBody(responsePtr));
+                            if (!string.IsNullOrEmpty(response))
+                            {
+                                var apiResponse = JsonConvert.DeserializeObject<ApiResponse>(response);
+
+                                _updateSampleRates(apiResponse.RateByService);
+                            }
+
+                            NativeInterop.ExporterResponse.Free(responsePtr);
```

Comment:
We probably need to wrap this in a try-catch to make sure we always call free?
```suggestion
                            try
                            {
                                string? response = Marshal.PtrToStringAnsi(NativeInterop.ExporterResponse.GetBody(responsePtr));
                                if (!string.IsNullOrEmpty(response))
                                {
                                    var apiResponse = JsonConvert.DeserializeObject<ApiResponse>(response);
    
                                    _updateSampleRates(apiResponse.RateByService);
                                }
                            }
                            catch(Exception ex)
                            {
                                _log.Error(ex, "An error occurred deserializing the response.");
                            }
                            finally
                            {
                                NativeInterop.ExporterResponse.Free(responsePtr);
                            }
```